hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/GrpcXceiverService.java

```diff
@@ -59,6 +59,7 @@ public void onNext(ContainerCommandRequestProto request) {
         } catch (Throwable e) {
           LOG.error("Got exception when processing"
                     + " ContainerCommandRequestProto {}", request, e);
+          isClosed.set(true);
```

Comment:
Should this be set after `responseObserver.onError(e);`?

Reply:
I think it's better to set it before that.  It should prevent another `onCompleted` call.

Reply:
From `StreamObersever.java`
```
Receives a terminating error from the stream.
May only be called once and if called it must be the last method called. In particular if an exception is thrown by an implementation of onError no further calls to any method are allowed.```

Reply:
The order looks fine to me